packages/vite/src/node/utils.ts

```diff
@@ -1194,3 +1194,16 @@ export const isNonDriveRelativeAbsolutePath = (p: string): boolean => {
   if (!isWindows) return p.startsWith('/')
   return windowsDrivePathPrefixRE.test(p)
 }
+
+export function joinUrlSegments(a: string, b: string): string {
+  if (!a || !b) {
+    return a + b
```

Comment:
nitpick, it may never happen 
```js
null + 'b'
undefined + 'b'
NaN + 'b'
```
maybe it is better?
```ts
return a || b || ''
```

Reply:
I was just concerned with the empty string case. I've updated it to your suggestion which I agree is a bit cleaner